### PR TITLE
b/a: do not set console in kernel command line for arm64

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
Do not set console in the kernel command line for arm64. The defaults
are better than setting ttyS0 as in x86, also this is our choice in
server images. The kernel will choose the preferred console depending
on whether dtb or ACPI is available:

1. When having a device tree, it will take the content of the
   /chosen/stdout-path node [1]. This node is properly filled in most
   of the cases.

2. If using ACPI tables, it will look for the console in the ACPI SPCR
   table [2,3]. This is different to what is done in x86 [4], as in that
   case the SPCR table is used only if earlycon is set.

[1] https://www.kernel.org/doc/Documentation/devicetree/bindings/chosen.txt
[2] https://docs.microsoft.com/en-us/windows-hardware/drivers/serports/serial-port-console-redirection-table
[3] https://elixir.bootlin.com/linux/v5.17/source/arch/arm64/kernel/acpi.c#L229
[4] https://elixir.bootlin.com/linux/v5.17/source/arch/x86/kernel/acpi/boot.c#L1615

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thank you

**Review by mardy — APPROVED** [review]
I was wondering if this would break i386 or other achitectures, but then I saw that our if/else block is such that only x86_64 and arm64 are supported. So, LGTM!

**Review by bboozzoo — APPROVED** [review]
LGTM

File: bootloader/assets/grub.go [hunk 1]
```diff
@@ -19,11 +19,21 @@
 
 package assets
 
+import (
+	"github.com/snapcore/snapd/arch"
+)
+
+var cmdlineForArch = map[string][]ForEditions{
+	"amd64": {{FirstEdition: 1, Snippet: []byte("console=ttyS0 console=tty1 panic=-1")}},
```
[the remark this answers is not in the document]
Comment (bboozzoo) [reply bootloader/assets/grub.go:27]: Thanks for spotting this!

**Comment by bboozzoo** [comment]
I've started a nested run, just to be extra sure we didn't break anything.

**Comment by bboozzoo** [comment]
The spread run completed. There is one error in nested Core 20 but it appears to be unrelated.

**Pull request merged**